h5py/_hl/filters.py

```diff
@@ -246,7 +246,7 @@ def rq_tuple(tpl, name):
     # End argument validation
 
     if (chunks is True) or \
-    (chunks is None and any((shuffle, fletcher32, compression, maxshape,
+    (chunks is None and any((shuffle, fletcher32, compression, maxshape and not len(external),
                              scaleoffset is not None))):
```

Comment:
While we're looking at this, I think it could be formatted more clearly.

```suggestion
    if (chunks is True) or (chunks is None and any((
            shuffle,
            fletcher32,
            compression,
            (maxshape and not len(external)),
            scaleoffset is not None,
    ))):
```

Other than this, the change LGTM.